### ISSUE TITLE
Is SlickNav's duplicate menu SEO friendly?

### ISSUE THREAD

**Issue by invest0r7** [ComputerWolf/SlickNav#105]
So an issue has come up that has me questioning if I should continue to use Slicknav. By default, Slicknav uses Javascript to copy your menu, which allows you to customize the functionality separately, which is very convenient.

However, with how Googlebot is starting to execute Javascript, it looks like it will basically see both menus, the natural raw coded one, and also the duplicate one executed via Javascript.

For example, viewing SlickNav's own site, you can see this issue when viewing it with Firebug or Developer Tools....You'll see the normal menu, and then right after the opening body tag you see the Javascript duplicate...

http://i.stack.imgur.com/GPVi4.jpg

So the issue is if this 2nd Javascript menu is seen and indexed by search spiders, which would essentially index duplicate content. I would have thought search engines would not see this javascript menu at all. I would think a search spider would only take into account the natural code, not Javascript-rendered code. But maybe that isn't so accurate anymore? Does anyone have any insight as to how search engines would treat this and if it's bad practice?

**Comment by ComputerWolf** [comment]
@invest0r7, this has been discussed heavily regarding many javascript plugins that use hidden data and how it is indexed and affects SEO. Most engines, such as Google, have become smart enough to recognize when `display:none` is used for enhancement purposes vs trying to manipulate your rankings. SlickNav does not place any data off-screen and instead uses `display:none` which is seen as the proper way to hide elements from view. While the data is still crawled, it is analyzed for intent as well.

This is touched on here: https://support.google.com/webmasters/answer/66353?hl=en
And other places: http://stackoverflow.com/questions/22313672/seo-display-none
https://productforums.google.com/forum/#!topic/webmasters/KMi4_wAjYw0

Basically, if you aren't being deceptive, you have nothing to worry about.

**Comment by invest0r7** [comment]
Thanks a ton for the reply Josh, much appreciated. So in essence, if they're smart enough to know what we're doing, simply serving a copy menu for mobile reasons, would it be safe to assume the engines wouldn't even count it as duplicate content/links? Would it be reasonable to think the engines know they're both the same menu, and as a result, treat them as one rather than two menus on the page?
